api/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -57,23 +62,23 @@ type IngressNodeProtocolConfig struct {
 	Protocol IngressNodeFirewallRuleProtocolType `json:"protocol"`
 
 	// tcp defines an ingress node firewall rule for TCP protocol.
-	// +optional
+	// +unionMember,optional
```

Comment:
These should be on separate lines, with this you've made your union member optional when the protocol is set